### PR TITLE
Register/Login UI and Validation

### PR DESCRIPTION
**Pull request by stevekroh**
- Integrates material-ui changes
- Adds validation to the Register page (Login changes are delayed)
- Switches configuration to node-config, so we can maintain multiple configurations without touching config.js

### CODE REVIEW

File: www/css/mod.css [hunk 1]
```diff
@@ -26,3 +26,8 @@
 .mdl-js-data-table {
   width: 100%;
 }
+
+/* This makes the login and register boxes look right. We'll fix this later */
```
Comment (JRJurman) [inline www/css/mod.css]: if this is for the login and register boxes, then let's not apply it to all `p` tags, can we use a classname?

File: www/js/containers/login-page.js [hunk 2]
```diff
@@ -1,51 +1,115 @@
...
       </div>
-    );
+      );
   }
 }
 
-export default LoginPage;
+function select( state ) {
+  return {
+    account: state.account
```
Comment (JRJurman) [inline www/js/containers/login-page.js]: is this used?

File: www/js/containers/register-page.js [hunk 3]
```diff
@@ -1,51 +1,176 @@
...
-        <div className="form-row">
-          <TextField type="password" floatingLabelText="Confirm Password"/>
-        </div>
-        <div className="form-row">
-          <RaisedButton onClick={this.onCancel.bind(this)}>
-            Cancel
+    const {account} = this.props;
+
+    let content;
+    if ( account.received && !account.registrationError ) {
```
Comment (JRJurman) [inline www/js/containers/register-page.js]: should we be using the Modal instead for notifications?

File: www/js/containers/login-page.js [hunk 4]
```diff
@@ -1,51 +1,115 @@
...
-                <Link to="register">Sign Up</Link> for Adventure Cycling
-              </CardText>
-              <CardText>
-                <a>Forgot Password?</a>
-              </CardText>
+      <div style={ styles.border }>
+        <div style={ styles.page }>
+          <img style={ styles.logo } src="./img/advc-big.jpg" />
+          <Paper style={ styles.block } zDepth={ 2 }>
+            <TextField hintText="Email"
```
Comment (JRJurman) [inline www/js/containers/login-page.js]: add `type="email"` to use the email keyboard

File: www/js/containers/register-page.js [hunk 5]
```diff
@@ -1,51 +1,176 @@
...
+          </p>
+          <p>
+            { account.registrationError }
+          </p>
+        </Paper>
+      );
+    } else {
+      const fields = [ {
+        hint: 'Email',
+        field: 'email'
```
Comment (JRJurman) [inline www/js/containers/register-page.js]: add `type: 'email'` to use email keyboard

File: package.json [hunk 6]
```diff
@@ -34,8 +43,10 @@
   "homepage": "https://github.com/bikelomatic-complexity/app#readme",
   "dependencies": {
     "blob-util": "^1.2.0",
+    "btc-models": "git+https://github.com/bikelomatic-complexity/btc-models.git",
     "docuri": "^4.2.1",
     "immutable": "^3.7.6",
+    "jquery": "^2.2.1",
```
Comment (JRJurman) [inline package.json]: do we have jquery listed twice?

File: www/css/mod.css (hunk 1 above)
Comment (JRJurman) [inline www/css/mod.css]: or an inline style?

**Comment by JRJurman** [comment]
image

File: www/js/containers/register-page.js [hunk 7]
```diff
@@ -1,51 +1,176 @@
...
     e.preventDefault();
 
-    this.props.history.pushState(null, 'login');
+    this.props.history.pushState( null, 'login' );
+  }
+
+  // Dispatch a registration request for all registration form fields. The
+  // fields are retreived by material-ui ref.
+  onRegister( e ) {
+    const attrs = [ 'email', 'password', 'username', 'first', 'last' ];
```
Comment (JRJurman) [inline www/js/containers/register-page.js]: should we maybe have a "confirm password"?

File: www/js/containers/login-page.js (hunk 2 above)
Comment (stevekroh) [inline www/js/containers/login-page.js]: It will eventually. Plus, we need access to dispatch

File: www/js/containers/register-page.js (hunk 3 above)
Comment (stevekroh) [inline www/js/containers/register-page.js]: More of a John question. I'll leave this in until we make a decision.

File: www/js/containers/register-page.js (hunk 7 above)
Comment (stevekroh) [inline www/js/containers/register-page.js]: Eventually. I'll postpone that for this PR though.

File: www/js/containers/login-page.js (hunk 2 above)
Comment (JRJurman) [inline www/js/containers/login-page.js]: you can just do a `connect()(LoginPage)`, and if we're not using account, let's just do that for now.

**Comment by JRJurman** [comment]
:tada: :cactus:

**Pull request merged**